src/adapter/src/coord/read_policy.rs

```diff
@@ -250,12 +250,15 @@ impl crate::coord::Coordinator {
                     let TimelineState { read_holds, .. } =
                         self.ensure_timeline_state(&timeline).await;
                     for (time, id_bundle) in &new_read_holds.holds {
-                        id_bundles.insert(Some(time.clone()), id_bundle.clone());
+                        id_bundles
+                            .entry(Some(time.clone()))
+                            .or_default()
+                            .extend(&id_bundle);
```

Comment:
Luckily this requires using multiple timelines to encounter a bug, so I think it's safe to say that no one encountered this in production.